src/Draco.Compiler/Syntax/Lexer.cs

```diff
@@ -356,7 +357,31 @@ private IToken LexString()
             // Count the number of required delimiters
             for (var i = 0; i < mode.ExtendedDelims; ++i)
             {
-                if (this.Peek(offset + i) != '#') goto not_escape_sequence;
+                if (this.Peek(offset + i) != '#')
+                {
+                    offset--;
+                    goto not_escape_sequence;
+                }
+            }
+
+            //interpolation
+            if (this.Peek(offset + mode.ExtendedDelims)=='{')
+            {
+                // Nothing lexed yet,we can return the end of string token
+                if (offset == 1)
+                {
+                    int count = mode.ExtendedDelims + 1;
+                    this.PushMode(ModeKind.Interpolation, 0);
+                    return IToken.From(TokenType.InterpolationStart, this.AdvanceWithText(count + 1));
+                }
+                else
+                {
+                    // This will only be interpolation in the next iteration, we just return what we have
+                    // consumed so far
+                    offset--;
```

Comment:
Again, it might be better to avoid state juggling and only advance offset truly when we are sure we can commit to it